python/cuml/test/test_random_forest.py

```diff
@@ -340,51 +358,83 @@ def test_rf_regression_float64(large_reg, datatype):
+
+def rf_classification(datatype, array_type, max_features, rows_sample,
+                      fixture):
+    X, y = fixture
     X = X.astype(datatype[0])
     y = y.astype(np.int32)
     X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=0.8,
                                                         random_state=0)
     X_test = X_test.astype(datatype[1])
 
+    handle, stream = get_handle(True, n_streams=1)
     # Initialize, fit and predict using cuML's
     # random forest classification model
-    cuml_model = curfc()
-    if type == 'dataframe':
+    cuml_model = curfc(max_features=max_features, rows_sample=rows_sample,
+                       n_bins=16, split_criterion=0,
+                       min_rows_per_node=2, seed=123, n_streams=1,
+                       n_estimators=40, handle=handle, max_leaves=-1,
+                       max_depth=16)
+    if array_type == 'dataframe':
         X_train_df = cudf.DataFrame.from_gpu_matrix(rmm.to_device(X_train))
         y_train_df = cudf.Series(y_train)
         X_test_df = cudf.DataFrame.from_gpu_matrix(rmm.to_device(X_test))
         cuml_model.fit(X_train_df, y_train_df)
-        cu_preds = cuml_model.predict(X_test_df,
-                                      predict_model="CPU").to_array()
+        cu_proba_gpu = np.array(cuml_model.predict_proba(X_test_df)
+                                .to_gpu_matrix())
+        cu_preds_cpu = cuml_model.predict(X_test_df,
+                                          predict_model="CPU").to_array()
+        cu_preds_gpu = cuml_model.predict(X_test_df,
+                                          predict_model="GPU").to_array()
     else:
         cuml_model.fit(X_train, y_train)
-        cu_preds = cuml_model.predict(X_test, predict_model="CPU")
+        cu_proba_gpu = cuml_model.predict_proba(X_test)
+        cu_preds_cpu = cuml_model.predict(X_test, predict_model="CPU")
+        cu_preds_gpu = cuml_model.predict(X_test, predict_model="GPU")
 
-    cu_acc = accuracy_score(y_test, cu_preds)
+    cu_acc_cpu = accuracy_score(y_test, cu_preds_cpu)
+    cu_acc_gpu = accuracy_score(y_test, cu_preds_gpu)
+    assert cu_acc_cpu == pytest.approx(cu_acc_gpu, abs=0.01, rel=0.1)
 
     # sklearn random forest classification model
     # initialization, fit and predict
-    if nrows < 500000:
-        sk_model = skrfc(max_depth=16, random_state=10)
+    if y.size < 500000:
+        sk_model = skrfc(n_estimators=40,
+                         max_depth=16,
+                         min_samples_split=2, max_features=max_features,
+                         random_state=10)
         sk_model.fit(X_train, y_train)
         sk_preds = sk_model.predict(X_test)
         sk_acc = accuracy_score(y_test, sk_preds)
-        assert cu_acc >= (sk_acc - 0.07)
+        sk_proba = sk_model.predict_proba(X_test)
+        assert cu_acc_cpu >= sk_acc - 0.07
+        assert cu_acc_gpu >= sk_acc - 0.07
+        check_predict_proba(cu_proba_gpu, sk_proba, y_test, 0.1)
```

Comment:
why rel=0.1? please can you add a comment explaining why you chose this particular value